From fb426aa31baeb8315f68be8c48987cb77252af74 Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Mon, 11 Mar 2019 16:42:14 +0000
Subject: [PATCH] TOOLS-2181 buildimage target should bail out early for
 non-zfs components Reviewed by: Trent Mick <trentm@gmail.com> Approved by:
 Trent Mick <trentm@gmail.com>

---
 tools/mk/Makefile.targ | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
index 6744fe8..33a33a4 100644
--- a/tools/mk/Makefile.targ
+++ b/tools/mk/Makefile.targ
@@ -425,8 +425,18 @@ endif
 
 endif
 
+#
+# We add a guard to ensure that users haven't invoked 'make buildimage'
+# by accident. It's better to bail out early with an clear error message
+# if a user has not declared that their component should $ENGBLD_USE_BUILDIMAGE.
+#
 .PHONY: buildimage
 buildimage: stamp-buildimage-prep release $(AGENTS:%=%-prebuilt)
+	if [[ "$(ENGBLD_USE_BUILDIMAGE)" != "true" ]]; then \
+		echo "This component is not configured to use buildimage."; \
+		echo "Set 'ENGBLD_USE_BUILDIMAGE=true' in the component Makefile otherwise."; \
+		exit 1; \
+	fi
 	mkdir -p $(ENGBLD_BITS_DIR)/$(NAME)
 	mkdir -p $(BUILDIMAGE_STAGEDIR)
 	cd $(BUILDIMAGE_STAGEDIR) ; \
-- 
2.21.0

